In [4]:
from math import factorial as fl

In [5]:
def C(k,n):
    return int(fl(n) / (fl(k) * fl(n-k)))

In [6]:
def A(k,n):
    return int(fl(n) / (fl(n-k)))

In [7]:
def P(n):
    return int(fl(n))

Задача 1:

Группа студентов изучает 10 различных дисциплин. Сколькими способами можно составить расписание на понедельник, если в этот день должно быть 4 разных занятия?

Так как порядок предметов имеет значение, используем формулу для размещения:

In [8]:
A(4,10)

5040

Ответ: 5040 способами.

Задача 2:

Из колоды в 52 карты вынимают случайным образом 4 карты. Найти число исходов, соответствующих тому, что был вытянут хотя бы один туз.

In [9]:
C(1,4)*C(3,48)+C(2,4)*C(2,48)+C(3,4)*C(1,48)+C(4,4)*C(0,48)

76145

Ответ: 76145 исходов.

Задача 3:
    
Семь человек рассаживаются наудачу на скамейке. Какова вероятность того, что два определённых человека будут сидеть рядом?

P(A) = (6 * 2 * 5!) / 7! = 2/7

In [10]:
(6 * 2 * fl(5)) / fl(7)

0.2857142857142857

Задача 4:
    
Из 60 вопросов, входящих в экзаменационные билеты, студент знает 50. Какова вероятность того, что среди трёх наугад выбранных вопросов студент знает: а) все? б) два?

In [29]:
C(3,50) / C(3,60)

0.5727644652250146

In [30]:
(C(2,50) * C(1,10)) / C(3,60)

0.35797779076563413

Ответ: а) 0.57, б) 0.36

Задача 5:
    
Бросается игральная кость. Пусть событие A - появление чётного числа, событие B - появление числа больше трёх. Являются ли эти события независимыми?

Ответ: события зависимые, так как появление четного числа влияет на появление числа больше трех (больше трех четных чисел два (4 и 6), а нечетное одно (5).

Задача 6:
    
Допустим, имеется некоторая очень редкая болезнь (поражает 0.1% населения). Вы приходите к врачу, вам делают тест на эту болезнь, и тест оказывается положительным. Врач говорит вам, что этот тест верно выявляет 99% больных этой болезнью и всего лишь в 1% случаев даёт ложный положительный ответ.
Вопрос: какова вероятность, что вы действительно больны ей?
Подсказка: вновь используйте формулу Байеса с раскрытием знаменателя с помощью формулы полной вероятности.

A - пациент болен редкой болезнью

B - тест на болезнь положительный

P(A|B) - вероятность болезни, в случае положительного теста (то, что нужно найти, так как по условию мы знаем, что тест положительный).

$$P(A|B) = P(B|A) \cdot P(A) / P(B)$$

P(B|A) = 0.99

P(A) = 0.001

$$P(B) = P(B|A) \cdot P(A) + P(B|\overline A) \cdot P(\overline A)$$

In [12]:
P_B = 0.001 * 0.99 + 0.999 * 0.01
print (P_B)

0.01098


P(A|B) = (0.99 * 0.001) / 0.01098

In [14]:
(0.99 * 0.001) / 0.01098

0.09016393442622951

Ответ: вероятность болезни 0.0902